src/main/java/uk/gov/pay/adminusers/resources/ServiceResource.java

```diff
@@ -227,28 +220,21 @@ public Response searchServices(JsonNode payload) {
                     @ApiResponse(responseCode = "201", description = "Created",
                             content = @Content(schema = @Schema(implementation = Service.class))),
                     @ApiResponse(responseCode = "409", description = "Gateway account IDs provided has already been assigned to another service"),
+                    @ApiResponse(responseCode = "400", description = "Service names must be one of 'en' (English) or 'cy' (Welsh)"),
                     @ApiResponse(responseCode = "500", description = "Invalid JSON payload")
             }
     )
-    public Response createService(JsonNode payload) {
-        LOGGER.info("Create Service POST request - [ {} ]", payload);
-        List<String> gatewayAccountIds = extractGatewayAccountIds(payload);
-        Map<SupportedLanguage, String> serviceNameVariants = getServiceNameVariants(payload);
+    public Response createService(@Valid CreateServiceRequest createServiceRequest) {
+        LOGGER.info("Create Service POST request - [ {} ]", createServiceRequest);
 
-        Service service = serviceServicesFactory.serviceCreator().doCreate(gatewayAccountIds, serviceNameVariants);
+        var nullableCreateServiceRequest = Optional.ofNullable(createServiceRequest);
+        Service service = serviceServicesFactory.serviceCreator().doCreate(
+                nullableCreateServiceRequest.map(CreateServiceRequest::gatewayAccountIds).orElse(List.of()),
+                nullableCreateServiceRequest.map(CreateServiceRequest::serviceName).orElse(Map.of()));
```

Comment:
Presumably the entire `createServiceRequest` is not going to be null but the list of gateway account IDs or map of languages to service names might be, so perhaps this would be better:

```java
Service service = serviceServicesFactory.serviceCreator().doCreate(
        Optional.ofNullable(createServiceRequest.gatewayAccountIds()).orElse(List.of()),
        Optional.ofNullable(createServiceRequest.serviceName()).orElse(Map.of()));
```

Reply:
I think this would be easier:
```
Service service = serviceServicesFactory.serviceCreator().doCreate(
        createServiceRequest == null ? List.of() : createServiceRequest.gatewayAccountIds(),
        createServiceRequest == null ? Map.of() : createServiceRequest.serviceName());
```
But am not very fussed here.

Reply:
⬆️ Actually my suggestion wouldn't work 🙈

Reply:
Maybe this would work:

```java
Service service = serviceServicesFactory.serviceCreator().doCreate(
        Optional.ofNullable(createServiceRequest).map(CreateServiceRequest::gatewayAccountIds).orElse(List.of()),
        Optional.ofNullable(createServiceRequest).map(CreateServiceRequest::serviceName).orElse(Map.of()));
```

It would get rid of the need to have the `nullableCreateServiceRequest` variable at the expense of an extra `Optional.ofNullable(…)` call (the cost of which is probably absolutely trivial).

This could be slightly rearranged to surface the actual data and types involved a bit more:

```java
List<String> gatewayAccountIds = Optional.ofNullable(createServiceRequest).map(CreateServiceRequest::gatewayAccountIds).orElse(List.of());
Map<SupportedLanguage, String> serviceNames = Optional.ofNullable(createServiceRequest).map(CreateServiceRequest::serviceName).orElse(Map.of());

Service service = serviceServicesFactory.serviceCreator().doCreate(gatewayAccountIds, serviceNames);
```